# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/BrowserChild.cpp
# Commit: f37913e7cb92
# Full Hash: f37913e7cb9295617b7e5a7ac5f462f29df583b4
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2021-08-18 15:53:06
# Regressor Bug: 1726063
# File Overlap Count: 1
# Description:
#   Bug 1726063 - Remove support for non-WR compositor initialization. r=jrmuizel,aosmond
#   
#   Depends on D122796
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D122798
# ==============================================================================

diff -r 7398bc391379 -r f37913e7cb92 dom/ipc/BrowserChild.cpp
--- a/dom/ipc/BrowserChild.cpp	Tue Aug 17 23:19:44 2021 +0000
+++ b/dom/ipc/BrowserChild.cpp	Tue Aug 17 23:19:44 2021 +0000
@@ -2878,51 +2878,14 @@
     mozilla::layers::PCompositorBridgeChild* aCompositorChild) {
   MOZ_ASSERT(aCompositorChild);
 
-  bool success = false;
-  if (mCompositorOptions->UseWebRender()) {
-    success = mPuppetWidget->CreateRemoteLayerManager(
-        [&](LayerManager* aLayerManager) -> bool {
-          MOZ_ASSERT(aLayerManager->AsWebRenderLayerManager());
-          nsCString error;
-          return aLayerManager->AsWebRenderLayerManager()->Initialize(
-              aCompositorChild, wr::AsPipelineId(mLayersId),
-              &mTextureFactoryIdentifier, error);
-        });
-  } else {
-    nsTArray<LayersBackend> ignored;
-    PLayerTransactionChild* shadowManager =
-        aCompositorChild->SendPLayerTransactionConstructor(ignored,
-                                                           GetLayersId());
-    if (shadowManager &&
-        shadowManager->SendGetTextureFactoryIdentifier(
-            &mTextureFactoryIdentifier) &&
-        mTextureFactoryIdentifier.mParentBackend !=
-            LayersBackend::LAYERS_NONE) {
-      success = true;
-    }
-    if (!success) {
-      // Since no LayerManager is associated with the tab's widget, we will
-      // never have an opportunity to destroy the PLayerTransaction on the next
-      // device or compositor reset. Therefore, we make sure to forcefully close
-      // it here. Failure to do so will cause the next layer tree to fail to
-      // attach due since the compositor requires the old layer tree to be
-      // disassociated.
-      if (shadowManager) {
-        static_cast<LayerTransactionChild*>(shadowManager)->Destroy();
-        shadowManager = nullptr;
-      }
-      NS_WARNING("failed to allocate layer transaction");
-    } else {
-      success = mPuppetWidget->CreateRemoteLayerManager(
-          [&](LayerManager* aLayerManager) -> bool {
-            ShadowLayerForwarder* lf = aLayerManager->AsShadowForwarder();
-            lf->SetShadowManager(shadowManager);
-            lf->IdentifyTextureHost(mTextureFactoryIdentifier);
-            return true;
-          });
-    }
-  }
-  return success;
+  return mPuppetWidget->CreateRemoteLayerManager(
+      [&](LayerManager* aLayerManager) -> bool {
+        MOZ_ASSERT(aLayerManager->AsWebRenderLayerManager());
+        nsCString error;
+        return aLayerManager->AsWebRenderLayerManager()->Initialize(
+            aCompositorChild, wr::AsPipelineId(mLayersId),
+            &mTextureFactoryIdentifier, error);
+      });
 }
 
 void BrowserChild::InitAPZState() {